OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -605,8 +605,10 @@ public void ResolveOrder(Actor self, Order order)
 				{
 					if (IsPlane)
 						self.QueueActivity(new ReturnToBase(self, Info.AbortOnResupply));
-					else
+					else if (!Info.RepairBuildings.Contains(order.TargetActor.Info.Name))
```

Comment:
Ok, after reading the code a bit more carefully I see your point and agree.  I suggest structuring and commenting it something like the following to make the (lack of coherant) logic clear:

```csharp
var targetReserved = Reservable.IsReserved(order.TargetActor);

// If the target for repair is reserved then drop the order
if (targetReserved && Info.RepairBuildings.Contains(order.TargetActor.Info.Name))
	return;

// If the target for reloading is reserved then automatically redirect to a different valid reload actor
// HACK: Repair and reload behaviour is inconsistent and one should be changed
if (targetReserved)
{
	// Existing content of if (Reservable.IsReserved(order.TargetActor)) branch
}
else
{
	// Existing content of else branch
}
```